src/tcgwars/logic/impl/gen4/SupremeVictors.groovy

```diff
@@ -1030,646 +1301,847 @@ public enum SupremeVictors implements LogicCardInfo {
+                }
+                flag = false
+              }
             }
           }
           move "Desperate Slap", {
             text "20+ damage. If Mr. Mime already has 5 or more damage counters on it, this attack does 20 damage plus 40 more damage."
             energyCost P, C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 20
+              if(self.numberOfDamageCounters >= 5) {
+                damage 40
+              }
             }
           }
-
         };
       case PARASECT_38:
-        return evolution (this, from:"Paras", hp:HP080, type:GRASS, retreatCost:1) {
+        return evolution (this, from:"Paras", hp:HP080, type:G, retreatCost:1) {
           weakness R, PLUS20
           move "Nutritional Support", {
-            text "Energy cards and attach them to any of your Pokémon in any way you like. Shuffle your deck afterward."
-            energyCost C, G
-            attackRequirement {}
+            text "Search your deck for up to 2 [G] Energy cards and attach them to any of your Pokémon in any way you like. Shuffle your deck afterward."
+            energyCost C
+            attackRequirement {
+              assert my.deck : "Your deck is empty"
+            }
             onAttack {
-              damage 0
+              attachEnergyFrom(type: G, max: 1, my.deck, my.bench.select())
+              attachEnergyFrom(type: G, max: 1, my.deck, my.bench.select())
+              shuffleDeck()
             }
           }
           move "Hibernation Spore", {
             text "40 damage. The Defending Pokémon is now Asleep. Flip 2 coins instead of 1 between turns. If either of them is tails, the Defending Pokémon is still Asleep."
             energyCost G, G, C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 40
+              afterDamage {
+                apply ASLEEP
+                def pcs = defending
+                targeted(pcs) {
+                  if (pcs.isSPC(ASLEEP)) {
+                    delayed {
+                      before ASLEEP_SPC, null, null, BEGIN_TURN, {
+                        flip "Asleep (Hibernation Spore)", 2, {}, {}, [2: {
+                          ef.unregisterItself(bg.em());
+                        }, 1:{
+                          bc "$ef.target is still asleep."
+                        }, 0:{
+                          bc "$pcs is knocked out by $thisMove."
+                          bc "$ef.target is still asleep"
+                        }]
+                        prevent()
+                      }
+                      after CLEAR_SPECIAL_CONDITION, pcs, {
+                        if (ef.types.contains(ASLEEP)) {
+                          unregister()
+                        }
+                      }
+                      after FALL_BACK, pcs, { unregister() }
+                      after KNOCKOUT, pcs, { unregister() }
```

Comment:
Would that work? I'm not familiar with how delegation works, but it seems like passing the effect to the static to be unregistered would be more complicated than than the other places where delegate is used.

Reply:
For delayed effects, the delegate is the `DelayedEffectGroovyImpl` itself. So everything you write in the implementation files inside of the (newest branch) `delayed` closure should be available to whatever you pass the delegate to in TcgStatics. You'd just call `delegate.unregister()` with the hard part being figuring out how to add `after`s and `before`s to something like that. However,  that isn't even hard, it just takes familiarity with Groovy. Let's use Escape Board as an example.

https://github.com/axpendix/tcgone-engine-contrib/blob/c5b758dcd607e6879fcf849f5f17f1ca7afe0d36/src/tcgwars/logic/impl/gen7/UltraPrism.groovy#L2952-L2970

If we change `eff2` to

```groovy
eff2 = delayed{
  delegate.before ASLEEP_SPC, self, null, RETREAT, {if(e.parentEvent.effect.retreater == self) prevent() }
  delegate.before PARALYZED_SPC, self, null, RETREAT, {if(e.parentEvent.effect.retreater == self) prevent() }
}
```

It will still work. `before` and `after` are just methods on the `DelayedEffectGroovyImpl` class and Groovy will parse this code just the same way as not including the `delegate.` parts for them. You can cast the delegate to get the IDE to recognize it and highlight it too.

![image](https://user-images.githubusercontent.com/634079/116324341-f6562480-a78d-11eb-9eb8-ec257490d3b9.png)

so basically all you have to do in the static is something like
```groovy
static unregisterCommon(def delegate, def target) {
  delegate.after FALL_BACK, target, { delegate.unregister() }
}
```

but for all the relevant ones.